common/src/main/java/org/apache/rocketmq/common/protocol/body/RegisterBrokerBody.java

```diff
@@ -40,4 +57,114 @@ public void setTopicConfigSerializeWrapper(TopicConfigSerializeWrapper topicConf
     public void setFilterServerList(List<String> filterServerList) {
         this.filterServerList = filterServerList;
     }
+
+    public byte[] encode(boolean compress) {
+
+        if (!compress) {
+            return encode();
+        }
+
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        DeflaterOutputStream outputStream = new DeflaterOutputStream(byteArrayOutputStream, new Deflater(Deflater.BEST_COMPRESSION));
```

Comment:
I think `Deflater.BEST_COMPRESSION` may be even slower than GZIP. Do we care about compression speed?